tests/next_tests/unit_tests/ffront_tests/foast_passes_tests/test_type_alias_replacement.py

```diff
@@ -42,3 +46,22 @@ def fieldop_with_typealias(
         foast_tree.body.stmts[0].value.left.func.id == expected
         and foast_tree.body.stmts[0].value.right.args[1].id == expected
     )
+
+
+def test_type_alias_replacement_astype_with_tuples():
```

Comment:
While from a user perspective the type alias feature is often used in conjunction with where on tuples, from a toolchain perspective the features are orthogonal (i.e. a bug in the type alias replacement has no effect on the where on tuples and vice-versa). I would suggest to remove this test and instead add an integration test (somewhere where `where` is tested) that ensures
```
    def where_on_tuples(
        a: gtx.Field[[TDim], float], b: gtx.Field[[TDim], float]
    ) -> tuple[gtx.Field[[TDim], float], gtx.Field[[TDim], float]]:
        return astype((a, b), float)
```
gives the expected result (i.e. the values are correct). If you want to go the full mile you can also add a unit test in `test_foast_to_itir.py` that ensures the generated ITIR is as expected. Please also add a test for nested tuples.

Reply:
I suppose you meant `astype` instead of `where`. The tests have been removed from the type alias tests and the integration tests have been added. I want to go the full mile, so the tests in `test_foast_to_itir.py` are to come.